internal/boxcli/midcobra/telemetry.go

```diff
@@ -201,3 +209,35 @@ func (m *telemetryMiddleware) trackEvent(evt *event) {
 			Set("shell", evt.Shell),
 	})
 }
+
+func identifyUser(segmentClient segment.Client, deviceID string) error {
+	username, err := openssh.GithubUsernameFromLocalFile()
+	if err != nil {
+		return err
+	}
+
+	userID := userIDFromGithubUsername(username)
+	if userID == "" {
+		// an empty userID means that we do not a github username saved
+		return nil
+	}
+
+	err = segmentClient.Enqueue(segment.Identify{
+		AnonymousId: deviceID,
+		UserId:      userID,
+		Traits:      segment.NewTraits().Set("githubUsername", username),
+	})
+	return errors.WithStack(err)
+}
+
+// userIDFromGithubUsername hashes the github username and produces a 64-char string as userId
+func userIDFromGithubUsername(username string) string {
+
+	const salt = "d6134cd5-347d-4b7c-a2d0-295c0f677948"
+	mac := hmac.New(sha256.New, []byte(salt))
```

Comment:
I'm not sure how much this matters, but the code we're using to hash the device ID has the salt in a different place than you do. See: https://github.com/denisbrodbeck/machineid/blob/f07179bb15a0bec252453468aa607c670f5c12b5/helper.go#L24 (`appID` is the salt)